driver/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java

```diff
@@ -41,6 +41,8 @@
     private int bufferOffset;
     private long currentPosition;
     private byte[] buffer = null;
+    private int markReadLimit;
```

Comment:
My reading of the InputStream Javadoc is that we can ignore the read limit:

     "However, the stream is not required to remember any data at all if more than <code>readlimit</code> bytes are read from the stream before <code>reset</code> is called."

but there's no reason I can see why it can't remember more than is asked for.